framework/include/loops/ComputeInitialConditionThread.h

```diff
@@ -33,4 +36,7 @@ class ComputeInitialConditionThread
 
   FEProblemBase & _fe_problem;
   THREAD_ID _tid;
+
+  /// @brief the names of target variables for which the initial conditions are applied
+  const std::set<std::string> _target_var_names;
```

Comment:
this is not used by the usual constructor so it s just left uninitialized. This is unexpected for anyone deriving from this class